networks/kim_cnn.py

```diff
@@ -0,0 +1,51 @@
+import torch
+import torch.nn as nn
+import torch.nn.functional as F
+
+from networks.base import BaseModel
+
+
+class KimCNN(BaseModel):
+    def __init__(self, config, embed_vecs):
+        super(KimCNN, self).__init__(config, embed_vecs)
+
+        self.filter_sizes = config.filter_sizes
+        emb_dim = embed_vecs.shape[1]
+        num_filter_maps = config.num_filter_maps
+        strides = config.strides
+
+        self.convs = nn.ModuleList()
+
+        for filter_size, stride in zip(self.filter_sizes, strides):
+            conv = nn.Conv2d(
+                in_channels=1,
+                out_channels=num_filter_maps,
+                kernel_size=(filter_size, emb_dim),
+                stride=(stride, emb_dim))
```

Comment:
This can is indeed a Conv1d operation.. Not sure if we want to do 2d here?? Need some discussion I think

Reply:
Though what we exactly do is 1D-convolution, we have to use Conv2D here because our input is 2D (doc_len, emb_size).
Conv1D can only handle 1D input (doc_len).

Reply:
Oh sorry, you're correct. We can transpose the input to treat emb_size as the number of input channels then use Conv1D.

Reply:
I found using Conv1d lead to significantly worse results.
Need further investigation...

Reply:
Solved. Should transpose the input for Conv1d.